src/main/java/gregtech/api/capability/impl/CommonFluidFilters.java

```diff
@@ -0,0 +1,114 @@
+package gregtech.api.capability.impl;
+
+import gregtech.api.capability.IFilter;
+import gregtech.api.unification.material.Material;
+import gregtech.api.unification.material.Materials;
+import gregtech.common.ConfigHolder;
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.FluidRegistry;
+import net.minecraftforge.fluids.FluidStack;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Common fluid filter implementations.
+ */
+public enum CommonFluidFilters implements IFilter<FluidStack> {
+    ALLOW_ALL {
+        @Override
+        public boolean test(@Nonnull FluidStack fluid) {
+            return true;
+        }
+
+        @Override
+        public int getPriority() {
+            return IFilter.noPriority();
+        }
+
+        @Override
+        public IFilter<FluidStack> negate() {
+            return DISALLOW_ALL;
```

Comment:
This looks like it may be a forward-reference. Is this allowed?

Reply:
It's inside a method, so it's probably fine.